packages/rest/src/rest.component.ts

```diff
@@ -94,6 +60,53 @@ export class RestComponent implements Component {
       Object.assign(apiSpec, {servers: config.openApiSpec.servers});
     }
     app.bind(RestBindings.API_SPEC).to(apiSpec);
+    app.server(RestServer);
+    this.bindBodyParsers(app);
+  }
+
+  /**
+   * Bind body parsers to the `RestServer`. This is needed because
+   * we bind `RequestBodyParser` as a singleton so that all body parser bindings
+   * need to be added to the same context.
+   *
+   * @param app Application instance
+   */
+  private bindBodyParsers(app: Application) {
+    const server = app.getSync<RestServer>(
```

Comment:
There may be more than a single (REST) server bound in the application. Before your change, all RestServer instance were obtaining the default list of built-in body parsers. With your change in place, non-default RestServers will end up with an empty list.

I feel we need to spend more time looking into different approaches for registrations of body parsers. I think it's a more generic problem that can be phrased as follows:
 - How to allow applications to configure bindings that are shared by all server instances?
 - How to allow applications to override list of bindings for a certain tag so that a selected server instance receives a different list of bindings (e.g. body parsers)?

To move your pull request forward and finish the cleanup of DI for singleton bindings, I am proposing to revert your changes related to body-parser configuration. To fix the problem surfaced by your cleanup, I am proposing to change the scope of body parsers from SINGLETON to TRANSIENT. IMO, body parsers should be lightweight classes that are cheap to instantiate, thus it should be ok to use transient scope for them.

Reply:
I reverted body parser changes and set the scope to be `TRANSIENT`.